### ISSUE TITLE
Inconsistent behaviour of GlobalMaxPooling2D Layer

### ISSUE THREAD

**Issue by J-Rocke** [keras-team/keras#11680]
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

The Problem is that GlobalMaxpooling2D returns different values depending on the previous layer beeing one of the outputs or not. My guess is it ignores the activation function of the previous layer.

I find the problem kind of hard to explain but i think the following simple example will show what i mean:

```
from tensorflow import keras
import numpy as np


input = keras.layers.Input(shape=(1, 1, 1))
conv = keras.layers.Conv2D(1, (1, 1), activation="sigmoid", kernel_initializer=keras.initializers.constant(2))(input)
max = keras.layers.GlobalMaxPool2D()(conv)

model = keras.models.Model(
    inputs=input,
    outputs=[conv, max],
)

p = model.predict(
    np.ones((1, 1, 1, 1))
)
print(p)
```

This prints:
> [array([[[[0.880797]]]], dtype=float32), 
> array([[0.880797]], dtype=float32)]

Important is to note that the value of the max output is roughly equal to 0.87 which is in between 0 and 1 as expected to be because of the previous sigmoid activation.

If i change the output of the model to be just the output of the GlobalMaxPool2D layer like shown here:

```
model = keras.models.Model(
    inputs=input,
    outputs=[max],
)
```
the output changes to 
> [[2.]]

which is not equal to the expected 0.87 and not even in the range [0,1] which is expected to be returned by the sigmoid activation of the conv2D layer.

It seems to me like the sigmoid activation of the previous Conv2D layer is ignored.

**Comment by gabrieldemarmiesse** [comment]
It seems like a TensorFlow bug to me. Or maybe there was a breaking change in TF?
